.github/workflows/base_tests.yml

```diff
@@ -0,0 +1,130 @@
+name: Base Tests
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+  workflow_call:
```

Comment:
Could you please elaborate more on why this is necessary.

Reply:
Ok, it seems it was necessary cause now the daily reuses the base tests. There was indeed duplication in those files. But one difference is that daily didn't use i`nstall_pinned`. This should continue as before.

Reply:
The reason I removed the `install_pinned` was because we were using both dependency install methods in the CI, and I tried to reduce them to one, as having multiple provides confusion and it might even lead to errors due to version differences.

Because of the reasons above, I still think it's a good idea to stick to just one method, but I'm certainly fine with either. Can you remind me why should it be the `install_pinned` one? Is that the default method for installing dependencies?

Reply:
The CI for the PRs use fixed versions in the .pinned files for stability purposes. The daily job uses the nimble resolution. Please, keep it like before.

Reply:
I am aware of the behaviour of those methods, but my question is about why are we using two different method for dependency management.

Reply:
The `.pinned` file behaves like a lock file - that wasn't supported before Nimble 0.14 - fixing the dependencies versions.

Reply:
Right. Does that mean it's supported now?

Reply:
Only for Nim 2, I guess. Not sure how well it works.

Reply:
We can try to use it instead of this .pinned file when we don't need to support Nim 1.6.x anymore.

Reply:
Alright.
Would it be alright to use this `.pinned` method for dependencies for the daily tests? Or is there some reason to keep using nimble's dep resolver?

Reply:
The .pinned file is used only by us to have a deterministic version of the dependencies. The libp2p.nimble file uses ranges or no dependency at all. This can be disruptive with newer deps versions being released. The daily job is like that because that's the likely context libp2p will be used as a dep.

Reply:
Feel free to document it with comments.

Reply:
In the daily workflow we test libp2p with newer deps versions without disrupting developers' PRs.

Reply:
I'm aware of the `.pinned` usages. 

But, as far as I know, wasn't the `nimble install` unreliable?